dlt-database/entity/0003-refactor_transaction_recipe/Account.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  Entity,
+  PrimaryGeneratedColumn,
+  Column,
+  ManyToOne,
+  JoinColumn,
+  OneToMany,
+  BaseEntity,
+} from 'typeorm'
+import { User } from '../User'
+import { Transaction } from '../Transaction'
+import { DecimalTransformer } from '../../src/typeorm/DecimalTransformer'
+import { Decimal } from 'decimal.js-light'
+import { AccountCommunity } from '../AccountCommunity'
+
+@Entity('accounts')
+export class Account extends BaseEntity {
+  @PrimaryGeneratedColumn('increment', { unsigned: true })
+  id: number
+
+  @ManyToOne(() => User, (user) => user.accounts, { cascade: ['insert'], eager: true }) // Assuming you have a User entity with 'accounts' relation
+  @JoinColumn({ name: 'user_id' })
+  user?: User
+
+  // if user id is null, account belongs to community gmw or auf
+  @Column({ name: 'user_id', type: 'int', unsigned: true, nullable: true })
+  userId?: number
+
+  @Column({ name: 'derivation_index', type: 'int', unsigned: true, nullable: true })
+  derivationIndex?: number
+
+  @Column({ name: 'derive2_pubkey', type: 'binary', length: 32, unique: true })
+  derive2Pubkey: Buffer
+
+  @Column({ type: 'tinyint', unsigned: true })
+  type: number
+
+  @Column({
+    name: 'created_at',
+    type: 'datetime',
+    precision: 3,
+    default: () => 'CURRENT_TIMESTAMP(3)',
+  })
+  createdAt: Date
+
+  @Column({ name: 'confirmed_at', type: 'datetime', nullable: true })
```

Comment:
In general i would suggest to define each datetime column with a precision of 3ms.
But especially in the context of transactions, contributions and accounts i think we MUST define a precision of 3 ms.
What do you think about?

And for the common columns like `created_at`, `updated_at`, `deleted_at`, ... we should try to use the typeorm annotations like @CreatedAtColumn, ...

Reply:
As far I can see, I use ms for all created_at columns and seconds for all confirmed_at
confirmed_at refer to the iota milestone for the corresponding transaction (also things like registerAddress for announcing/creating new user and accounts) and the iota milestone has only seconds precision. 

Yeah we could try it.

Reply:
We could try it in backend. I have revisited my code, actually I don't need set creation_at automatically,
because accounts and users and communities and transactions are created elsewhere and I get this date either from backend or blockchain. 
So I updated the entities in 0003-refactor_transaction_recipe and remove all CURREN_TIMESTAMP defaults.

Reply:
But you need to use instead of the CURRENT_TIMESTAMP the annotation @CreationDateColumn, right? 
Otherwise there is no automatism for initializing the default time.

Reply:
I don't use the automatism for initializing default time, I don't need default time. 
In every use case I get creation_at from another source.